### PR TITLE
Add performance test for OP

### PR DESCRIPTION
**Pull request by laoliu97**
### 包含算子：
torch.nn.functional.cosine_similarity
torch.cumsum
torch.cumprod
torch.nn.functional.conv_transpose2d
torch.diag
torch.diagonal
torch.div
torch.dot
torch.nn.functional.dropout
torch.eq
torch.erf
torch.erfc
torch.erfinv
torch.expm1
torch.eye
torch.flatten
torch.flip
torch.floor
torch.fmod
torch.nn.functional.fold
torch.gather
torch.nn.functional.glu
torch.gt
torch.nn.functional.grid_sample
torch.hann_window

### 统计覆盖情况
image

### CODE REVIEW

**Comment by laoliu97** [comment]
# 统计覆盖情况
image

File: python/oneflow/test/modules/test_cosine_similarity.py [hunk 1]
```diff
@@ -58,6 +58,14 @@ def test_cosine_similartiy_module_with_nonequal_dim_data(test_case):
...
         output = cos(a, b)
         return output
+    '''
+    @profile(torch.nn.functional.cosine_similarity)
+    def profile_cosine_similarity(test_case):
+        input1 = torch.ones(100,128)
+        input2 = torch.ones(100,128)
+        torch.nn.functional.cosine_similarity(input1, input2)
+        torch.nn.functional.cosine_similarity(input1, input2, dim=0) 
+    '''
```
Comment (daquexian) [inline python/oneflow/test/modules/test_cosine_similarity.py:68]: 这个注释原因是什么

Comment (laoliu97) [reply python/oneflow/test/modules/test_cosine_similarity.py:68]: 这部分添加profile后测试中出现了[assert error](https://github.com/Oneflow-Inc/oneflow/issues/8881#issuecomment-1229682453),本来是添加在issue中的，可能当时网络不好没有comment成功，抱歉

Comment (daquexian) [reply python/oneflow/test/modules/test_cosine_similarity.py:68]: 好的，可以不采用注释掉的方式，而是在 62 行上面加 @unittest.skip(reason="https://github.com/Oneflow-Inc/oneflow/issues/8881#issuecomment-1229682453")，这样便于追溯上下文

File: python/oneflow/test/modules/test_deconv2d.py [hunk 2]
```diff
@@ -937,6 +937,22 @@ def test_deconv2d_group_with_random_data(test_case):
...
+        torch.nn.functional.conv_transpose2d(inputs, weights_64c)
+        torch.nn.functional.conv_transpose2d(inputs, weights_64c, stride=2)
+        torch.nn.functional.conv_transpose2d(inputs, weights_64c, stride=2, padding=1)
+        torch.nn.functional.conv_transpose2d(inputs, weights_64c, stride=2, padding=1, output_padding=1)
+        torch.nn.functional.conv_transpose2d(inputs, weights_64c, stride=2, padding=1, output_padding=1, bias=torch.ones(64))
+        torch.nn.functional.conv_transpose2d(inputs, weights_5x5_64c)
+        torch.nn.functional.conv_transpose2d(inputs, weights_5x5_64c, stride=2)
+        torch.nn.functional.conv_transpose2d(inputs, weights_5x5_64c, stride=2, padding=1)
+        torch.nn.functional.conv_transpose2d(inputs, weights_5x5_64c, stride=2, padding=1, output_padding=1)
+        torch.nn.functional.conv_transpose2d(inputs, weights_5x5_64c, stride=2, padding=1, output_padding=1, bias=torch.ones(64))
```
Comment (daquexian) [inline python/oneflow/test/modules/test_deconv2d.py:954]: 可以调查下 3x3 反卷积和 5x5 反卷积的常见 padding 各是什么，防止测试了很多操作而实际没人会这么用

Comment (laoliu97) [reply python/oneflow/test/modules/test_deconv2d.py]: 已修改，根据实际情况分别用上采样2倍和4倍（k=4,s=2,p=1和k=8,s=4,p=2）进行测试

File: python/oneflow/test/modules/test_cosine_similarity.py (hunk 1 above)
Comment (laoliu97) [reply python/oneflow/test/modules/test_cosine_similarity.py]: 已修改

File: python/oneflow/test/modules/test_deconv2d.py [hunk 3]
```diff
@@ -937,6 +937,16 @@ def test_deconv2d_group_with_random_data(test_case):
         y = m(x)
         return y
 
+    @profile(torch.nn.functional.conv_transpose2d)
+    def profile_conv_transpose2d(test_case):
+        inputs = torch.ones(16, 3, 128, 128)
+        weights_64c = torch.ones(3, 64, 4, 4)
```
Comment (daquexian) [inline python/oneflow/test/modules/test_deconv2d.py:943]:
```suggestion
        weights_4x4_64c = torch.ones(3, 64, 4, 4)
```

另外 input 的 channel 是 3 是不是也不符合常见情况

此外可以再加一个包含 output_padding 的测试

Comment (laoliu97) [reply python/oneflow/test/modules/test_deconv2d.py]: 已修改，增加上采样3倍的操作

**Review by daquexian — APPROVED** [review]

**Review by doombeaker — APPROVED** [review]
